test/perf.jl

```diff
@@ -0,0 +1,91 @@
+using BenchmarkTools
+using Setfield
+using Test
+using InteractiveUtils
+
+struct AB{A,B}
+    a::A
+    b::B
+end
+
+function lens_set_a(obj, val)
+    @set obj.a = val
+end
+
+function hand_set_a(obj, val)
+    AB(val, obj.b)
+end
+
+function lens_set_ab(obj, val)
+    @set obj.a.b = val
+end
+
+function hand_set_ab(obj, val)
+    a = AB(obj.a.a, val)
+    AB(a, obj.b)
+end
+
+function lens_set_a_and_b(obj, val)
+    o1 = @set obj.a = val
+    o2 = @set o1.b = val
+end
+
+function hand_set_a_and_b(obj, val)
+    AB(val, val)
+end
+
+function uniquecounts(iter)
```

Comment:
This helper counts how often an element occurs in a collection. 
```julia
julia> uniquecounts([1,2,1,3,2,2,2])
Dict{Int64,Int64} with 3 entries:
  2 => 4
  3 => 1
  1 => 2
```
I feel like reinventing the whell here, @tkf do you know if such functionality is somewhere in Base or stdlib?

Reply:
I don't know.  But maybe not, as there is `countmap` in non-std package `StatsBase` http://juliastats.github.io/StatsBase.jl/latest/counts.html#StatsBase.countmap.